src/affexpr.jl

```diff
@@ -31,6 +31,21 @@ Base.zero(a::GenericAffExpr) = zero(typeof(a))
 Base.one( a::GenericAffExpr) =  one(typeof(a))
 Base.copy(a::GenericAffExpr) = GenericAffExpr(copy(a.vars),copy(a.coeffs),copy(a.constant))
 
+"""
+    value(a::GenericAffExpr, map::Function)
+
+Evaluate `a` given the value `map(v)` for each variable `v`.
+"""
+function value(a::GenericAffExpr{T, V}, map::Function) where {T, V}
```

Comment:
Why not use a `VariableToValueMap` instead of a generic function? No need to infer the result type of `map`.

Reply:
You cannot get the `VariableToValueMap` because you do not have a reference to the `model`.

Reply:
I'm confused. I mean define ``function value(a::GenericAffExpr{T, V}, map::VariableToValueMap{S}) where {T, V, S}``.

Reply:
Oh, this is too generic with respect to the variable type. Leave this for now, but we're going to change this later when we have a coherent story for dealing with the different result types.